src/efi.rs

```diff
@@ -105,6 +110,70 @@ impl Efi {
     }
 }
 
+/// Convert a nul-terminated UTF-16 byte array to a String.
+fn string_from_utf16_bytes(slice: &[u8]) -> String {
+    // For some reason, systemd appends 3 nul bytes after the string.
```

Comment:
Hmm, messy.  But I wonder if we should care about this?  Maybe instead we return a `Vec<u8>` since in the end all we care about is if it starts with the ASCII string `systemd` right?  

OK as is though.

Reply:
You can't just send an array of bytes (which you already have from reading the file contents) back since they're be misinterpreted as UTF-8 since every other byte is `nul`. That's why you have to convert it to `Vec<u16>` and parse it as UTF-16. I don't believe there's any spec that says you have to write UTF-16 strings to EFI variables. I think that's just a convention systemd follows since that's how other vendors do it. I think I read a comment in the systemd code that they append 3 `nul` bytes so that in case they start parsing in the middle of the buffer they'll still end up with a valid C string. However, stripping an odd numbered trailing byte seems like good practice.

When I wrote roughly the same code in Python where I didn't have to change types, I walked backwards from the end of the buffer until I found no more pairs of `nul` bytes. Here I let `U16CString` handle that part.

Reply:
Yes...hm.  What about going the other way, and converting from `systemd` in UTF-8 to UTF-16, and then using https://doc.rust-lang.org/std/primitive.slice.html#method.starts_with ?